### PR TITLE
Implement the create() method of sdk interface

### LINKED ISSUE #56: Implement Create() Method of SDK Interface
Provide implementation for the Create() method defined on SDK interface and include relevant unit tests.

### PR DESCRIPTION
**Pull request by KarimElghamry**
* modify the create method on the sdk interface to throw
  annotator and stream exceptions
* implement the create method on the DefaultSdk
* implement related unit tests

Fix #56

Signed-off-by: Karim Elghamry <karimelghamry@gmail.com>

### CODE REVIEW

File: src/test/java/com/alvarium/SdkTest.java [hunk 1]
```diff
@@ -69,4 +69,30 @@ public void createShouldReturnSameData() {
...
+          .getType(), sdkInfo.getSignature());
+    }
+
+    // init logger and sdk
+    final Logger logger = LogManager.getRootLogger();
+    Configurator.setRootLevel(Level.DEBUG);
+    final Sdk sdk = new DefaultSdk(annotators, sdkInfo, logger);
+
+    // init property bag and data
+    final PropertyBag properties = new ImmutablePropertyBag(new HashMap<String, Object>());
```
Comment (tsconn23) [inline src/test/java/com/alvarium/SdkTest.java:92]: This is kind of a bummer in that we have to force the developer to init and pass in a PropertyBag when we aren't going to do anything with it. Right now, I'm unsure if this will be the norm or a corner case.

A couple questions
1.) Would it make sense to have overloaded SDK methods (`sdk.create(data)` in this case) where no additional properties are required?
2.) Is there any value in having the test above initialize a TLS annotator through the config and then passing a dummy TLS session into `create()` via the PropertyBag to show how that works?

Comment (KarimElghamry) [reply src/test/java/com/alvarium/SdkTest.java:92]: 1) Overloading the sdk methods is a good approach in my opinion. The only problem that I can think of regarding this approach is that we'll have to instantiate property bags inside all of the sdk methods so that we can pass them to the underlying annotators. We'll also have to add three more methods to the sdk interface. Another approach would be changing the `PropertyBag` param to [`Optional`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) and check whether the value is present or not before passing it to the annotators. Either way, I think we can do these changes in a future PR in parallel with the work done in the example app.

2) I've already written a [unit test](https://github.com/project-alvarium/alvarium-sdk-java/blob/main/src/test/java/com/alvarium/annotators/TlsAnnotatorTest.java#L41) for the TLS annotator that demonstrates how a user can pass in an `SSLSocket` through the `PropertyBag`. I was aiming to test the sdk methods fully using mock implementations to reduce the dependency in unit tests. I believe that adding TLS annotator tests on top of the sdk methods won't add significant value as it's already done in a separate unit test. I think we can fully demonstrate the general workflow inside the example app.
